src/widgets/table/table.rs

```diff
@@ -1484,7 +1364,264 @@ mod tests {
                 .rows(vec![])
                 .footer(Row::new(vec!["h", "i"]))
                 .column_spacing(0);
-            assert_eq!(table.get_columns_widths(10, 0), &[(0, 5), (5, 5)])
+            assert_eq!(table.get_columns_widths(10, 0), [(0, 5), (5, 5)])
+        }
+
+        #[test]
+        fn spacers_priority_unconstrained() {
+            fn test(
+                highlight_spacing: HighlightSpacing,
+                selection: Option<usize>,
+                expected: Buffer,
+            ) {
+                let table = Table::default()
+                    .rows(vec![Row::new(vec!["hello", "world"])])
+                    .highlight_spacing(highlight_spacing)
+                    .highlight_symbol(">>>")
+                    .column_spacing(0);
+                let area = Rect::new(0, 0, 15, 3);
+                let mut buf = Buffer::empty(area);
+                if let Some(s) = selection {
+                    let mut state = TableState::default().with_selected(Some(s));
+                    StatefulWidget::render(table, area, &mut buf, &mut state);
+                    assert_buffer_eq!(buf, expected);
+                } else {
+                    Widget::render(table, area, &mut buf);
+                    assert_buffer_eq!(buf, expected);
+                }
+            }
+
+            // no highlight_symbol rendered ever
```

Comment:
It wouldn't hurt to pull these out to individual methods

Reply:
I think these tests all fall into the same category, i.e. how is space allocated when there's lots of room to do so. Thoughts?

Reply:
I'm going to leave this the way it is for now.

Reply:
This was the lowest priority comment - it's only a thing because each assertion has a comment, which would make a perfect name for the test. That said, the extra comments on each individual assertion suggest that there's enough complexity to generally do this I think.